crates/sparrow-physical/src/step.rs

```diff
@@ -32,45 +33,59 @@ pub struct Step {
     pub kind: StepKind,
     /// Inputs to this step.
     pub inputs: Vec<StepId>,
-    /// The schema for this step.
-    pub schema: SchemaRef,
+    /// The data type produced by this step.
+    pub data_type: DataType,
```

Comment:
Or `result_type/output_type`, so as to not have to look at the docs to clarify that it's not the "input type to this step"?

Reply:
Yeah, I like that. Any concern that it may be confusing to have "output_type" of the step and "output_type" of each expression? I guess not.